src/Orleans.Runtime/Streams/QueueBalancer/LeaseBasedQueueBalancer.cs

```diff
@@ -174,41 +171,44 @@ public override IEnumerable<QueueId> GetMyQueues()
             await this.leaseProvider.Release(LeaseCategory, queuesToGiveUp.Select(queue => queue.AcquiredLease).ToArray());
             //remove queuesToGiveUp from myQueue list after the balancer released the leases on them
             this.myQueues.RemoveRange(0, queueCountToRelease);
-            this.logger.Info($"ReleaseLeasesToMeetResponsibility: released {queueCountToRelease} queues, current queue Count: {this.myQueues.Count}");
+            this.logger.Info($"Released leases for {queueCountToRelease} queues");
+            this.logger.LogInformation($"I now own leases for {this.myQueues.Count} of an expected {this.minimumResponsibility} to {this.maximumResponsibility} queues.");
         }
 
-        private Task AcquireLeaseToMeetMaxResponsibilty(object state)
+        private Task AcquireLeaseToMeetMaxResponsibility(object state)
         {
-            return AcquireLeasesToMeetExpectation(this.maximumRespobsibility);
+            return AcquireLeasesToMeetExpectation(this.maximumResponsibility);
         }
 
         private Task AcquireLeasesToMeetMinResponsibility()
         {
-            return AcquireLeasesToMeetExpectation(this.minimumResponsibilty);
+            return AcquireLeasesToMeetExpectation(this.minimumResponsibility);
         }
 
         private async Task AcquireLeasesToMeetExpectation(int expectedTotalLeaseCount)
         {
             int maxAttempts = 5;
             int attempts = 0;
             int leasesToAquire = expectedTotalLeaseCount - this.myQueues.Count;
-            this.logger.Info($"AcquireLeasesToMeetExpectation : Try to acquire {leasesToAquire} queues");
-            while (attempts ++ <= maxAttempts && leasesToAquire > 0)
+            if (leasesToAquire <= 0) return;
+            while (attempts ++ < maxAttempts && leasesToAquire > 0)
             {
+                this.logger.LogDebug($"I have {this.myQueues.Count} queues.  Trying to acquire {leasesToAquire} queues to reach {expectedTotalLeaseCount}");
+                leasesToAquire = expectedTotalLeaseCount - this.myQueues.Count;
                 //select new queues to acquire
-                var expectedQueues = this.queueSelector.NextSelection(leasesToAquire, this.myQueues.Select(queue=>queue.QueueId).ToList()).ToList();
-                var leaseRequests = expectedQueues.Select(queue => new LeaseRequest() {
+                List<QueueId> expectedQueues = this.queueSelector.NextSelection(leasesToAquire, this.myQueues.Select(queue=>queue.QueueId).ToList()).ToList();
+                IEnumerable<LeaseRequest> leaseRequests = expectedQueues.Select(queue => new LeaseRequest() {
                     ResourceKey = queue.ToString(),
                     Duration = this.options.LeaseLength
                 });
-                var results = await this.leaseProvider.Acquire(LeaseCategory, leaseRequests.ToArray());
+                AcquireLeaseResult[] results = await this.leaseProvider.Acquire(LeaseCategory, leaseRequests.ToArray());
```

Comment:
why are we so explicit about the type. I thought using var is the suggested way

Reply:
It's a trade off, maintainability vs readability.  Historically, the convention on the team has been to use explicit type if it's not obvious from the call.  This is a carryover from 343 coding conventions.
So something like:

    var foo = services.GetService<IFoo>()

is fine because it's obvious what the type is, but:

    var thingys = foo.GetAllThingys()

is less desirable, because it's not clear to a reviewer what thingy is.  We've never strongly enforced this, and its not clear if this convention should carry forward with the new team.  I still follow the convention mostly out of habit.